sycl/include/sycl/types.hpp

```diff
@@ -10,2207 +10,8 @@
-#error "Undefine __SYCL_UOP macro"
-#endif
-#define __SYCL_UOP(UOP, OPASSIGN)                                              \
-  vec &operator UOP() {                                                        \
-    *this OPASSIGN vec_data<DataT>::get(1);                                    \
-    return *this;                                                              \
-  }                                                                            \
-  vec operator UOP(int) {                                                      \
-    vec Ret(*this);                                                            \
-    *this OPASSIGN vec_data<DataT>::get(1);                                    \
-    return Ret;                                                                \
-  }
-
-  __SYCL_UOP(++, +=)
-  __SYCL_UOP(--, -=)
-#undef __SYCL_UOP
-
-  // operator~() available only when: dataT != float && dataT != double
-  // && dataT != half
-  template <typename T = DataT>
-  typename std::enable_if_t<!std::is_floating_point_v<vec_data_t<T>> &&
-                                (!IsUsingArrayOnDevice && !IsUsingArrayOnHost),
-                            vec>
-  operator~() const {
-    vec Ret{(typename vec::DataType) ~m_Data};
-    if constexpr (std::is_same_v<Type, bool>) {
-      Ret.ConvertToDataT();
-    }
-    return Ret;
-  }
-  template <typename T = DataT>
-  typename std::enable_if_t<!std::is_floating_point_v<vec_data_t<T>> &&
-                                (IsUsingArrayOnDevice || IsUsingArrayOnHost),
-                            vec>
-  operator~() const {
-    vec Ret{};
-    for (size_t I = 0; I < NumElements; ++I) {
-      Ret.setValue(I, ~getValue(I));
-    }
-    return Ret;
-  }
-
-  // operator!
-  template <typename T = DataT, int N = NumElements>
-  EnableIfNotUsingArray<vec<T, N>> operator!() const {
-    return vec<T, N>{(typename vec<DataT, NumElements>::DataType) !m_Data};
-  }
-
-  // std::byte neither supports ! unary op or casting, so special handling is
-  // needed. And, worse, Windows has a conflict with 'byte'.
-#if (!defined(_HAS_STD_BYTE) || _HAS_STD_BYTE != 0)
-  template <typename T = DataT, int N = NumElements>
-  typename std::enable_if_t<std::is_same_v<std::byte, T> &&
-                                (IsUsingArrayOnDevice || IsUsingArrayOnHost),
-                            vec<T, N>>
-  operator!() const {
-    vec Ret{};
-    for (size_t I = 0; I < NumElements; ++I) {
-      Ret.setValue(I, std::byte{!vec_data<DataT>::get(getValue(I))});
-    }
-    return Ret;
-  }
-
-  template <typename T = DataT, int N = NumElements>
-  typename std::enable_if_t<!std::is_same_v<std::byte, T> &&
-                                (IsUsingArrayOnDevice || IsUsingArrayOnHost),
-                            vec<T, N>>
-  operator!() const {
-    vec Ret{};
-    for (size_t I = 0; I < NumElements; ++I)
-      Ret.setValue(I, !vec_data<DataT>::get(getValue(I)));
-    return Ret;
-  }
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
I would vote for adding a comment here about how to proceed with bugfix/further changes to `vec` class. If I understand correctly, all new API/ABI breaking changes should only go into `vec_new`, whilst generic implementation bugfix should be done in both versions